src/models/account.js

```diff
@@ -168,12 +171,29 @@ module.exports = (db) => {
             address: to
           })
         })
+      },
+
+      setWalletAddress: async function (newAddress) {
+        this.walletAddress = newAddress
+        await this.saveAsync().catch(e => {
+          console.error(`Error while setting public wallet address of Account object:\nNewAddress: ${newAddress}\nlException:${JSON.stringify(e)}`)
+          return Promise.reject(e)
+        })
       }
     },
 
     hooks: {
       beforeSave: function () {
         const now = new Date()
+
+        // If our walletAddress is set, we need to make sure it's a valid wallet address before saving
+        if(typeof this.walletAddress !== 'undefined' && this.walletAddress !== null) {
+          if(!StellarSdk.StrKey.isValidEd25519PublicKey(this.walletAddress)) {
+            this.walletAddress = null
+            return Promise.reject(new Error('BAD_PUBLIC_WALLET_ADDRESS'))
```

Comment:
I definitely have some confusion over where to use promises and where not to use promises in the app. Should I not be using Promises explicitly at all? Just using async/await and throwing errors where I would otherwise reject?